ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -316,8 +316,28 @@ extension SignalProducerType {
 	///
 	/// Returns a Disposable which can be used to interrupt the work associated
 	/// with the Signal, and prevent any future callbacks from being invoked.
-	public func start(error error: (E -> ())? = nil, completed: (() -> ())? = nil, interrupted: (() -> ())? = nil, next: (T -> ())? = nil) -> Disposable {
-		return start(Event.sink(next: next, error: error, completed: completed, interrupted: interrupted))
+	public func startWithNext(error error: (E -> ())? = nil, completed: (() -> ())? = nil, interrupted: (() -> ())? = nil, next: (T -> ())? = nil) -> Disposable {
```

Comment:
> I'm not sure. I think keeping it clean at the call site could be worth an somewhat un-intuitive argument ordering.

This is really why I don't like (2) because it doesn't work when you have more than one closure argument. I think I'm back to the (3) camp with single closure argument and use `start` if you want to handle multiple.